bench/src/main/scala/bench/GenBench.scala

```diff
@@ -0,0 +1,104 @@
+package bench
+
+import java.util.concurrent.TimeUnit
+import org.openjdk.jmh.annotations._
+
+import org.scalacheck.Gen
+import org.scalacheck.rng.Seed
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Array(Mode.AverageTime))
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+class GenBench {
+
+  // if we only benchmark with one seed, we might get lucky (or
+  // unlucky) for a particular generator. by benchmarking with 10
+  // different seeds (constant between runs) we reduce the chance of
+  // extreme luckiness or unluckness. that said, a generator that hits
+  // a 100x slow down 1% of the time will likely cause problems for
+  // this benchmarking strategy.
+  val preseeds: List[Long] =
```

Comment:
There could be some utility to using `@Param` and `@Setup` from JMH in combination with `Random.nextLong`.  It would take some investigation.  Seed literals probably have a benefit.

Reply:
I've done this in the past. The big down side is that the benchmarks become somewhat more non-deterministic.

I could be convinced to use a parameter determining how many deterministic seeds we use (e.g. 10, 50, 100).

Reply:
Yeah, determinism in benchmarks with literals is a benefit.